File: patch_pkg.py
import json

with open('package.json', 'r') as f:
    pkg = json.load(f)

deps_to_remove = [
    '@google/genai',
    'better-sqlite3',
    'date-fns',
    'dotenv',
    'express',
    'gray-matter',
    'react-markdown',
    'remark-gfm',
    'vite'
]

for dep in deps_to_remove:
    if 'dependencies' in pkg and dep in pkg['dependencies']:
        del pkg['dependencies'][dep]

dev_deps_to_remove = [
    '@types/express',
    'tsx'
]

for dep in dev_deps_to_remove:
    if 'devDependencies' in pkg and dep in pkg['devDependencies']:
        del pkg['devDependencies'][dep]

with open('package.json', 'w') as f:
    json.dump(pkg, f, indent=2)
    f.write('\n')

File: patch_pkg.js
const fs = require('fs');
const pkg = JSON.parse(fs.readFileSync('package.json', 'utf8'));

const depsToRemove = [
    '@google/genai',
    'better-sqlite3',
    'date-fns',
    'dotenv',
    'express',
    'gray-matter',
    'react-markdown',
    'remark-gfm',
    'vite'
];

for (const dep of depsToRemove) {
    if (pkg.dependencies && pkg.dependencies[dep]) {
        delete pkg.dependencies[dep];
    }
}

const devDepsToRemove = [
    '@types/express',
    'tsx'
];

for (const dep of devDepsToRemove) {
    if (pkg.devDependencies && pkg.devDependencies[dep]) {
        delete pkg.devDependencies[dep];
    }
}

fs.writeFileSync('package.json', JSON.stringify(pkg, null, 2) + '\n');
